### ISSUE TITLE
Major changes in topology graph since last version

### ISSUE THREAD

**Issue by wsw2016** [Interoperable-data/ERA-Ontology-3.1.0#11]
In the previous version of ERA-vocabulary the topology graph was defined by two classes NetElement and NetRelation. With them we could model topologies, where the NetElements are connected by their ends.
 
![graph_1](https://github.com/Interoperable-data/ERA-Ontology-3.1.0/assets/22107843/99dc5883-e42d-45ec-af37-facaffdfaff1)

In the new version an additional object is introduced “NonLinearElement”, which can be placed on a LinearElement and link to a connection (era:Navigability). Now the following graph is representable.
![graph_2](https://github.com/Interoperable-data/ERA-Ontology-3.1.0/assets/22107843/d75dbbc1-3578-4a0e-b6b7-833a29676986)

Is it by intension? Both, RailML3 and Eulynx dataprep models support only the first kind of the topology graph.

**Comment by Airy59** [comment]
RTM1.1, RSM1.x (consumed by EULYNX DP and, in altered form, by railML(R)3.x) all allow both representations.

But a track level (MICRO level), and by convention, only linear elements are used for the topology graph.

This is to ensure that navigabilities are transitive and can be used to determine possible paths between an origin and a destination.

The same convention may well apply to the ERA Ontology. That is ERA decision of course.

**Comment by wsw2016** [comment]
As long as the era:from and era:to do not explicitly refer to an end of a LinearElement, we can connect the middle of one LinearElement with an end of another LinearElement, and the position of this linking can be defined via NonLinearElement.

If this would be made impossible by referencing era:from to LinearElementEnd (or so), the reason for the NonLinearElement is unclear: it is not used for the graph definition. It is using the graph and so, can be defined in some other package.

**Comment by Airy59** [comment]
Indeed, the nonlinear element can be dispensed of. In the ERA proposal, it is used as the "carrier" of navigabilities: possible, but not necessary.

**Comment by sixdiamants** [comment]
The notion of orientation of the linear element seems to have disappeared. This was given by the axiom that a net relation connects the two linear elements head-to-tail, head-to-head etc. In other words, the net relation being attached to the start/end of a linear element. 

Did I miss an episode ?

**Comment by Airy59** [comment]
The orientation should not disappear indeed, lest the pathfinding become impossible using topology alone.

**Comment by Airy59** [comment]
Here's further analysis with (counter) examples

**Comment by Airy59** [comment]
See pull request. You also have a copy of the modified ontology (ontology.ttl), a sample track layout (pdf), and the corresponding instantiation of topology objects (example.ttl) available on [https://cdm.ovh/ERA_310_topology_fork](url)

**Comment by nicolasgatez** [comment]
another problem arises when considering a more macro level, where the pathfinding requests are done from OP to OP, not from SOL to SOL. this means that in this case it is the set of relations that acts as nodes and the linear elements that act as edges. 

The nonlinear  netelements were introduced to avoid changing the logic between levels and allow mixed levels.
